### ISSUE TITLE
Broke rules after optimization

### ISSUE THREAD

**Issue by unlok** [jakubpawlowicz/clean-css#616]
There are two independent files:
```css
.class
{
    border: 2px solid #fc0;
    border-right-width: 0;
}
```
and
```css
.another-class
{
    border: 2px solid #fc0;
}
```

The files can used independently and together. And when it together, result is broken rules:
```css
.class{border-right-width:0}
.another-class,.class{border:2px solid #fc0}
```

First occurs setting of sub-rule `border-right-width` but then occurs settings of `border` and it override sub-rule.

It must be as follows:
```css
.another-class,.class{border:2px solid #fc0}
.class{border-right-width:0}
```

**Comment by coreyworrell** [comment]
See #619 as well.

**Comment by jakubpawlowicz** [comment]
This is a bug :+1:

**Comment by tavriaforever** [comment]
:+1:

**Comment by jakubpawlowicz** [comment]
The fix is on 3.3 branch so you can give it a try.

If it's all good I'll release it on Tuesday when back from a short break.

**Comment by tavriaforever** [comment]
Wow, thanks! We are going to try)

**Comment by jakubpawlowicz** [comment]
It's released as 3.3.7 now.
